File: human-or-not/Assets/Scripts/Network/Packet/Packet.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public enum PacketId
{
    // S: Server's packet, C: Client's packet
    C_CreateRoom = 1,
    C_EnterRoom = 2,
    C_LeaveRoom = 3,
    S_CreateRoom = 4,
    S_EnterRoom = 5,
    S_LeaveRoom = 6,
    S_Spawn = 7,
    S_Despawn = 8,
}

public abstract class Packet
{
    public ushort Protocol;
}

#region Server's packet
[Serializable]
public class S_CreateRoom : Packet
{
    public PlayerInfo user;
    public string roomId;
}

[Serializable]
public class S_EnterRoom : Packet
{
    public PlayerInfo user;
    public string roomId;
}

[Serializable]
public class S_Spawn : Packet
{
    public PlayerInfo[] users;
}
#endregion

#region Client's packet
[Serializable]
public class C_CreateRoom : Packet
{
    public string userName;
    public C_CreateRoom()
    {
        Protocol = (ushort)PacketId.C_CreateRoom;
    }
}

[Serializable]
public class C_EnterRoom : Packet
{
    public string playerName;
    public string roomId;

    public C_EnterRoom()
    {
        Protocol = (ushort)PacketId.C_EnterRoom;
    }
}
#endregion

File: server/src/packet.js
const PACKET_ID = {
  C_CreateRoom: 1,
  C_EnterRoom: 2,
  S_PlayerOrder: 3,
  S_BroadcastEnterRoom: 4,
};

class Packet {
  constructor(protocol) {
    this.Protocol = protocol;
  }
}

class C_CreateRoom extends Packet {
  constructor() {
    super(PACKET_ID.C_CreateRoom);
  }
}

class C_EnterRoom extends Packet {
  constructor(roomId) {
    super(PACKET_ID.C_EnterRoom);
    this.roomId = roomId;
  }
}

class S_PlayerOrder extends Packet {
  constructor(playerId, order) {
    super(PACKET_ID.S_PlayerOrder);

    this.playerId = playerId;
    this.order = order;
  }
}

class S_BroadcastEnterRoom extends Packet {
  constructor(playerId) {
    super(PACKET_ID.S_BroadcastEnterRoom);
    this.playerId = playerId;
  }
}

module.exports = {
  PACKET_ID,
  S_PlayerOrder,
  C_CreateRoom,
  C_EnterRoom,
  S_BroadcastEnterRoom,
};
